crates/storage/nippy-jar/src/compression/zstd.rs

```diff
@@ -0,0 +1,175 @@
+use crate::{compression::Compression, NippyJarError};
+use serde::{Deserialize, Serialize};
+use std::{
+    fs::File,
+    io::{Read, Write},
+};
+use zstd::bulk::{Compressor, Decompressor};
+
+#[derive(Debug, Default, PartialEq, Serialize, Deserialize)]
+pub enum ZstdState {
+    #[default]
+    PendingDictionary,
+    Ready,
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+/// Zstd compression structure. Supports a compression dictionary per column.
+pub struct Zstd {
+    /// State. Should be ready before compressing.
+    pub(crate) state: ZstdState,
+    /// Uses custom dictionaries to compress data.
+    pub(crate) use_dict: bool,
+    /// Max size of a dictionary
+    pub(crate) max_dict_size: usize,
+    /// List of column dictionaries.
+    pub(crate) raw_dictionaries: Option<Vec<Vec<u8>>>,
+    /// Number of columns to compress.
+    columns: usize,
+}
+
+impl Zstd {
```

Comment:
Should the columns be named/labeled, i.e. make it a map?

Reply:
I'm not sure I follow...